### ISSUE TITLE
Feature Request: Create comparer for strings

### ISSUE THREAD

**Issue by skarllot** [diegofrata/Generator.Equals#52]
Create comparer for strings:

```csharp
[StringEquality(StringComparison.OrdinalIgnoreCase)]
public string Name { get; set; }
```

**Comment by diegofrata** [comment]
Somehow missed this issue for a while. I think this is generally a good idea! Happy to take a PR for it.

In the meantime this could be accomplished with:

```csharp
[CustomEquality(typeof(StringComparer), nameof(StringComparer.OrdinalIgnoreCase))] 
```

**Comment by diegofrata** [comment]
This is done as part of #55. Should be available on 3.1.0.
